test/stmt/switch_stmt2.swift

```diff
@@ -136,3 +136,17 @@ func non_fully_covered_switch(x: Int) -> Int {
   }
   return x
 }
+
+// Do not crash if another switch statement follows a fallthrough.
+func fallthrough_not_last(i: Int) {
+  switch i {
+  case 1:
+    fallthrough
+    switch i {
```

Comment:
What does SILGen do in this case?

Reply:
Treats it like "goto $next-case" after mandatory passes warn that it's unreachable.

Reply:
Can you see if there are tests for this? ie, some code after a fallthrough is diagnosed as unreachable and deleted. If not, it would be nice to add some since it might easily break in the future.

Reply:
I don't see any tests that test specifically for this behavior. A correction: It's SILGen that is emitting the diagnostic based on the insertion point being cleared after a `fallthrough`.

Before we add a test, it might be good to ensure that this is the behavior we want. It seems reasonable to just treat code following a fallthrough (other than perhaps a #if that evaluates to false) as an error. @jckarter, do you know if the current behavior was intentional?

Reply:
Having said that, perhaps an error is too paternal and inconvenient for people experimenting.

Reply:
Yeah, we don't error on code after a return either. Please add a SILGen test specifically for the fallthrough case in a follow-up commit.

Reply:
I'll take a look at that after investigating why the `CSGen.cpp` change [here](https://github.com/apple/swift/pull/17497) broke a source compatibility test.